pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -278,3 +300,73 @@ func isNotFoundError(err error) bool {
 	}
 	return false
 }
+
+func (r *Reconciler) instanceExistsInPool(instanceLink string, pool string) (bool, error) {
+	// Get target pool
+	tp, err := r.computeService.TargetPoolsGet(r.projectID, r.providerSpec.Region, pool)
+	if err != nil {
+		return false, fmt.Errorf("unable to get targetpool: %v", err)
+	}
+
+	// Check if instance is in target pool already
+	if tp.Instances != nil {
+		for _, link := range tp.Instances {
+			if instanceLink == link {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+type poolProcessor func(instanceLink, pool string) error
+
+func (r *Reconciler) processTargetPools(poolFunc poolProcessor) error {
+	if r.providerSpec.TargetPools != nil && len(r.providerSpec.TargetPools) > 0 {
+		instanceSelfLink := fmtInstanceSelfLink(r.projectID, r.providerSpec.Zone, r.machine.Name)
+		for _, pool := range r.providerSpec.TargetPools {
+			err := poolFunc(instanceSelfLink, pool)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) addInstanceToTargetPool(instanceLink string, pool string) error {
+	inPool, err := r.instanceExistsInPool(instanceLink, pool)
```

Comment:
what happens if you don't check and just add?

Reply:
nothing.  If you just add repeatedly, it will just succeed.  Technically speaking, we might not need to check if it exists, but I think having the check might make feedback to the user more consumable if something fails; eg, did the get fail, or the add-instance fail (Did user specify a target pool that doesn't exist)?  Ideally, we only do this once during create, and for all but the initial time this runs, we'll only be doing 1 api call: 2 calls first time, get and add, every other time, just get.